scripts/conversion_script.py

```diff
@@ -292,32 +274,39 @@ def cleanup(required_files):
     not something that was downloaded by the script.
     """
     for required_file in required_files:
-        if not required_file.is_file_present and os.path.exists(required_file.path):
+        if os.path.exists(required_file.path):
             print(
                 "Removing the file '%s' as it was previously downloaded."
                 % required_file.path
             )
             os.remove(required_file.path)
-            continue
-
-        print(
-            "File '%s' was present on the system before the execution. Skipping the removal."
-            % required_file.path
-        )
+        _create_or_restore_backup_file(required_file)
 
 
-def verify_required_files_are_present(required_files):
-    """Verify if the required files are already present on the system."""
-    print("Checking if required files are present on the system.")
-    for required_file in required_files:
-        # Avoid race conditions
-        try:
-            print("Checking for file %s" % required_file.path)
-            with open(required_file.path, mode="r") as handler:
-                required_file.sha512_on_system = hashlib.sha512(handler.read())
-                required_file.is_file_present = True
-        except (IOError, OSError):
-            required_file.is_file_present = False
+def _create_or_restore_backup_file(required_file):
+    """
+    Either creates or restores backup files (rename in both cases).
+    """
+    suffix = ".backup"
+    try:
+        if os.path.exists(required_file.path + suffix):
+            print("Restoring backed up file %s." % (required_file.path))
+            os.rename(required_file.path + suffix, required_file.path)
+            return
+        if os.path.exists(required_file.path):
+            print(
+                "File %s already present on system, backing up to %s."
+                % (required_file.path, required_file.path + suffix)
+            )
+            os.rename(required_file.path, required_file.path + ".backup")
+    except (IOError, OSError) as exception:
+        # pylint: disable=raise-missing-from
+        # This is expected and we want to control the flow with ProcessError
+        # instead of re-raising the exception.
+        raise ProcessError(
+            message="An unexpected error occured during restore or backup of required file.",
+            report=str(exception),
+        )
```

Comment:
Might be better to let the exception fall through and be caught in the main exception.

Reply:
ah right, that would be easier .. probably though about it too much :sweat_smile: